From f4efae870df9f8e6f237bb51fb8d238984cf9346 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@freescale.com>
Date: Fri, 25 Jan 2013 14:56:04 -0600
Subject: [PATCH 411/486] ENGR00241860 [DISPLAY] - Need to implement blanking
 for the hardware composer

The hardware composer in Android Jelly Bean 4.2 now requires a display
blanking function.

Signed-off-by: Oliver Brown <oliver.brown@freescale.com>
---
 mx5x/hwcomposer/hwcomposer.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index a8272fc..c401a70 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -176,6 +176,19 @@ static int hwc_eventControl(struct hwc_composer_device_1* dev, int disp, int eve
 
 static int hwc_blank(struct hwc_composer_device_1* dev, int disp, int blank)
 {
+    struct hwc_context_t* ctx = (struct hwc_context_t*)dev;
+
+    if (!ctx || disp < 0 || disp >= HWC_NUM_DISPLAY_TYPES) {
+        return 0;
+    }
+
+    int fb_blank = blank ? FB_BLANK_POWERDOWN : FB_BLANK_UNBLANK;
+    int err = ioctl(ctx->m_mainfb_fd, FBIOBLANK, fb_blank);
+    if (err < 0) {
+        ALOGE("blank ioctl failed");
+        return -errno;
+    }
+
     return 0;
 }
 
-- 
1.8.0

